WordPress/Classes/ViewRelated/NUX/SigninHelpers.swift

```diff
@@ -10,44 +11,133 @@ import WordPressComAnalytics
     private static let AuthenticationEmailKey = "AuthenticationEmailKey"
 
 
-    //MARK: - Helpers for presenting the signin flow
+    // MARK: - AB test related methods
+
+
+    /// Loads the mixpanel experiments (if necessary) and shows a variant of the sign in flow.
+    /// A call is made to join mixpanel experiements, passing a callback to show
+    /// the relevant sign in variant. To compensate for latency, a brief timer is
+    /// scheduled to do the same task.  Regardless of whether the timer or the
+    /// callback is performed first the second call will be ignored by design.
+    ///
+    class func loadABTestThenShowSigninController() {
+        guard let rootController = WordPressAppDelegate.sharedInstance().window.rootViewController else {
+            assertionFailure("Oh where oh where could my rootviewcontroller have gone?")
+            return
+        }
+
+        if useNewSigninFlow() {
+            // We know we're using a varient so proceeed without wait.
+            SigninHelpers.showSigninFromPresenter(rootController, animated: false, thenEditor: false)
+            return
+        }
+
+        // Keep showing the launch screen until we know which signin varient we want.
+        let storyboard = UIStoryboard(name: "Launch Screen", bundle: NSBundle.mainBundle())
+        let controller = storyboard.instantiateInitialViewController()!
+        let navController = NUXNavigationController(rootViewController: controller)
+        navController.toolbarHidden = true
+
+        rootController.presentViewController(navController, animated: false, completion: nil)
+
+        // Load A/B tests.
+        Mixpanel.sharedInstance().joinExperimentsWithCallback {
+            SigninHelpers.showSigninControllerForABTest()
+        }
+
+        NSTimer.scheduledTimerWithTimeInterval(2, target: SigninHelpers.self, selector: #selector(SigninHelpers.showSigninControllerForABTest), userInfo: nil, repeats: false)
+    }
+
+
+    /// Displays a variant of the sign in flow.
+    /// Also checks if the visible view controller is already one of our test
+    /// variants and if so returns early.
+    ///
+    class func showSigninControllerForABTest() {
+        guard let rootController = WordPressAppDelegate.sharedInstance().window.rootViewController else {
+            assertionFailure(" Oh where oh where could he be?")
```

Comment:
As much as I love cute messages, I'd like for them to at least be informative. Can you add a mention about the rootController not being available please?